src/blockdata/script.rs

```diff
@@ -809,44 +809,57 @@ impl_index_newtype!(Builder, u8);
 #[cfg(feature = "serde")]
 impl<'de> serde::Deserialize<'de> for Script {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'de>,
+    where D: serde::Deserializer<'de>,
     {
-        use std::fmt::Formatter;
-        use hashes::hex::FromHex;
+        if deserializer.is_human_readable() {
+            use hashes::hex::FromHex;
```

Comment:
A use inside of an if seems very unusual, is this intentional?

Reply:
I just put it there because it was inside the method in the beginning and the trait is needed only in that if branch...
If you don't like it, I would probably remove it and use the trait function directly where it's needed (`Vec::from_hex(v)` -> `hex::FromHex::from_hex(v)`)